src/m365/purview/commands/threatassessment/threatassessment-list.ts

```diff
@@ -0,0 +1,108 @@
+import { Logger } from '../../../../cli/Logger';
+import GlobalOptions from '../../../../GlobalOptions';
+import { odata } from '../../../../utils/odata';
+import GraphCommand from '../../../base/GraphCommand';
+import commands from '../../commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  type?: string;
+}
+
+class PurviewThreatassessmentListCommand extends GraphCommand {
+  private static readonly allowedTypes: string[] = ['mail', 'file', 'emailFile', 'url'];
+
+  public get name(): string {
+    return commands.THREATASSESSMENT_LIST;
+  }
+
+  public get description(): string {
+    return 'Get a list of threat assessment';
+  }
+
+  public defaultProperties(): string[] | undefined {
+    return ['id', 'contentType', 'category'];
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initValidators();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        type: typeof args.options.type !== 'undefined'
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-t, --type [type]', autocomplete: PurviewThreatassessmentListCommand.allowedTypes
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+
+        if (args.options.type && PurviewThreatassessmentListCommand.allowedTypes.indexOf(args.options.type) < 0) {
+          return `${args.options.type} is not a valid type. Valid types are ${PurviewThreatassessmentListCommand.allowedTypes.join(', ')}`;
+        }
+        return true;
+      }
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    if (this.verbose) {
+      logger.logToStderr(`Retrieving list of threat assessment`);
+    }
+
+    let url = `${this.resource}/v1.0/informationProtection/threatAssessmentRequests`;
+    if (args.options.type) {
+      url += '';
+    }
+
+    try {
+      const items = await odata.getAllItems<any>(url, 'minimal');
+      if (args.options.type) {
```

Comment:
Can't we we use the `$filter=` query parameter instead?

Reply:
seems not, unless i'm doing something wrong? 
![image](https://user-images.githubusercontent.com/35696168/221203815-29a0e149-f49f-4508-8a89-46393bb47288.png)

Reply:
Ignore my comment. Apparently the `contentType` propertie does not support `emailFile` 😄

Reply:
Hmmm, the endpoint is currently giving me 401 errors. Let's wait a bit, I want to try it out as well :)

Reply:
> Hi @nicodecleyre, is the endpoint working on your end? On my end it still gives me 401 errors.

Same here 
![image](https://user-images.githubusercontent.com/35696168/229114154-e60bdba6-e77b-4ca7-8430-97be059822ed.png)